### ISSUE TITLE
exec_auth spawns the command synchronously 

### ISSUE THREAD

**Issue by feloy** [kubernetes-client/javascript#2044]
**Describe the bug**

exec_auth spawns the command synchronously. 

When the exec command sends information to stdout/stderr to inform the user what to do to authenticate, the program is blocked waiting indefinitely, because the spawn is done synchronously.

 

** Client Version **
1.0.0-rc7

** Server Version **

Azure 1.29.9

**To Reproduce**

- Start an Azure cluster with 'Microsoft Entra ID authentication with Kubernetes RBAC' authn & authz
- get the credentials
- do not execute any request to the cluster, so the token is not cached (or you can remove it: `rm ~/.kube/cache/kubelogin/yourID.json`
- make a request to the API
- the program is blocked executing the `kubelogin` app, which is blocked, waiting for the user to authenticate

**Expected behavior**

The call should be done async
bonus: the information provided by kubelogin should be displayed to the user

** Example Code**

```
import { CoreV1Api, KubeConfig } from "@kubernetes/client-node";

const kc = new KubeConfig();
kc.loadFromDefault();

const k8sApi = kc.makeApiClient(CoreV1Api);
const result = k8sApi.listNamespace();
```

**Environment (please complete the following information):**
 - OS: MacOS
 - NodeJS Version: v20.15.1
 - Cloud runtime: Azure

**Additional context**

**Comment by brendandburns** [comment]
Happy to take PRs for this. We should detect if there is an active try and send `interactive` or not to the exec plugin.

**Comment by feloy** [comment]
I have a poc working, I'll look at this `interactive` thing and propose a PR. 
Thanks

**Comment by feloy** [comment]
@brendandburns I can see that `kubelogin` accepts a `--login interactive` option, which makes the user experience better, as the login is opened on the browser when necessary (the fact that spawn is synchronous is blocking the GUI, which is still annoying).

But I cannot see how `kubelogin` can work with the auto-detected interactive mode. I tried to modify my kubeconfig to add `interactiveMode: IfAvailable` and keep it with `--login devicecode` and am trying with `kubectl`, but I still get a code, not interactive session

**Comment by feloy** [comment]
Fixed by #2046
